Lib/test/test_operator.py

```diff
@@ -474,6 +474,30 @@ def __getitem__(self, other): return 5  # so that C is a sequence
         self.assertEqual(operator.ixor     (c, 5), "ixor")
         self.assertEqual(operator.iconcat  (c, c), "iadd")
 
+    def test_iconcat_without_getitem(self):
+        operator = self.module
+        class X: pass
+
+        msg = "'X' object can't be concatenated"
+        with self.assertRaisesRegex(TypeError, msg):
+            operator.iconcat(X(), X())
```

Comment:
I think it is better to test some objects which support addition, for example
```suggestion
            operator.iconcat(1, 0.5)
```